distribution/lib/Standard/Base/0.2.32-SNAPSHOT/src/Data/Vector.enso

```diff
@@ -393,12 +445,41 @@ type Vector
 
              [1, 2, 3].to_text == "[1, 2, 3]"
     to_text : Text
-    to_text =
-        if this.length == 0 then "[]" else
-            if this.length == 1 then "[" + (this.at 0 . to_text) + "]" else
-                folder = str -> ix -> str + ", " + (this.at ix).to_text
-                tail_elems = 1.up_to this.length . fold "" folder
-                "[" + (this.at 0 . to_text) + tail_elems + "]"
+    to_text = this.generic_to_text "[" ", " "]"
+
+    ## PRIVATE
+       A generic utility which concatenates textual representations of the
+       vector elements, separated by the `separator` and appends the provided
+       `prefix` and `suffix`.
+    generic_to_text : Text -> Text -> Text -> Text
+    generic_to_text prefix separator suffix =
```

Comment:
there's a `Vector.join` for that, why not add `prefix` and `suffix` to it?

Reply:
`join` works for a Vector of Texts, this one works for an arbitrary Vector, by applying `to_text`.
But I guess I can re-use join and simply `.map .to_text` in the `short_display_text` function.

Reply:
Removed `generic_to_text` and switched to a map+join.